### PR TITLE
Videos do not reload 

### PR DESCRIPTION
**Pull request by nathangthomas**
This PR does the following: 
- Adds a namespaced route for new bookmarks controller
- Adds namespaced bookmark controller 
- Moves previous bookmark controller functionality to new controller
- Improves user experience by adding bookmarkVideo function that incorporates fetch to enable a user to bookmark a video without reloading the page and starting the video over
- Refactors bookmark video button to route to new action

**Functionality works on localhost however testing in this PR needs work. I need to do further research into the problems I am having. There are notes in the test file if you would like to take a look.**

### CODE REVIEW

File: app/controllers/api/v1/bookmarks_controller.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+class Api::V1::BookmarksController < ApplicationController
+  skip_before_action :verify_authenticity_token
+
+  def create
+    user_video = UserVideo.new(user_video_params)
+    if current_user.user_videos.find_by(video_id: user_video.video_id)
+#flash messages will no longer work due to fetch call. Need a solution to this.
```
Comment (fentontaylor) [inline app/controllers/api/v1/bookmarks_controller.rb]: If you want to remember this, can you make it an issue and delete this comment?

**Comment by fentontaylor** [comment]
Really great work on this. I admire your dedication to tackling and solving a completely unknown problem.

File: spec/features/user/bookmarking_a_video_does_not_reload_video_playing_spec.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(user)
+
+    @tutorial = create(:tutorial)
+    create(:video, tutorial: @tutorial)
+
+  end
+
+  it 'when i visit a tutorial and click_on the bookmark button the page does not reload and the video is added to my bookmarks' do
+    visit(tutorial_path(@tutorial.id))
+    click_button("Bookmark")
```
Comment (nancylee713) [inline spec/features/user/bookmarking_a_video_does_not_reload_video_playing_spec.rb]: I did encounter a similar issue when I was working on 'importing youtube playlist'. Based on my research, the problem might have to do with capybara needing to wait for all ajax requests to finish (i.e., `Capybara.default_max_wait_time`). However, the cause of this problem might lie elsewhere.

**Comment by nancylee713** [comment]
Remarkable work on this extension!

**Pull request merged**